src/libraries/System.Net.Quic/tests/FunctionalTests/QuicConnectionTests.cs

```diff
@@ -56,6 +56,149 @@ public async Task AcceptStream_ConnectionAborted_ByClient_Throws()
                     Assert.Equal(ExpectedErrorCode, ex.ErrorCode);
                 });
         }
+
+        private static async Task DoWrites(QuicStream writer, int writeCount)
+        {
+            for (int i = 0; i < writeCount; i++)
+            {
+                await writer.WriteAsync(new byte[1]);
+            }
+        }
+
+        private static async Task DoReads(QuicStream reader, int readCount)
+        {
+            for (int i = 0; i < readCount; i++)
+            {
+                int bytesRead = await reader.ReadAsync(new byte[1]);
+                Assert.Equal(1, bytesRead);
+            }
+        }
+
+        [Theory]
+        // [InlineData(0)]  // Fails with TimeoutException -- needs investigation
```

Comment:
Isn't it because since you don't do any write, the stream start won't kick off and nothing gets actually sent over the wire?

Reply:
It's probably related to that. But it still seems like stream should get failed, even if it hasn't sent anything on the wire yet.

Reply:
This is indeed because the stream won't be created on the wire until you write at least 1 byte to it, so serverConnection.AcceptStreamAsync() never finishes (and so semaphore is never released, and so connection is not closed).

Funny thing is if you call CloseAsync without waiting for semaphore, this would actually flush stream creation event and peer's AcceptStreamAsync would finish.

I wonder if there's a way to manually flush connection events. If not, I don't think there's much we can do.

Reply:
Yeah, you are right of course -- the timeout comes from the AcceptStreamAsync.

> I wonder if there's a way to manually flush connection events. If not, I don't think there's much we can do.

I think that FlushAsync on a stream that hasn't been started on the wire should force it to be started on the wire.

That would simplify a few things, like the stream construction logic we use for the Stream Conformance Tests, where we currently have to send and receive a byte in order to force stream creation before using it for the tests.

But that's not something we need to tackle right now, I think. Besides, it gets in to the broader discussion about send buffering and flushing that we are punting for 6.0.

Reply:
All that said, for now I am just going to remove these "0" variations, since they end up behaving quite differently than the others. If you think it's worthwhile to have tests for this, let's figure out how to add them later; but they are probably separate tests.